### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by piezol**
Homework Podstawy C++ #2
Authors: @piezol, @mefox256

### CODE REVIEW

File: module2/homework/nwdNww/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,29 @@
 #pragma once
-
+#include <math.h>
+#include <algorithm>
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if(lhs==0 || rhs==0){
```
Comment (kontrybutor) [inline module2/homework/nwdNww/nwdNww.hpp:5]: Please add space before and after ```==``` operator.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,29 @@
 #pragma once
-
+#include <math.h>
```
Comment (kontrybutor) [inline module2/homework/nwdNww/nwdNww.hpp:2]: Please use ```<cmath>``` as  includes with .h postfix are rather used in C code.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+    return lhs;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if(lhs == 0 || rhs == 0){
+        return 0;
+    }
+    lhs = abs(lhs);
```
Comment (kontrybutor) [inline module2/homework/nwdNww/nwdNww.hpp:24]: Please be consistent with yourself :) In NWD function you used ```std::abs```. I don't see the reason why it should be omitted here.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+    rhs = abs(rhs);
+    int result;
+    result = lhs/NWD(lhs, rhs) * rhs;
+    return result;
```
Comment (kontrybutor) [inline module2/homework/nwdNww/nwdNww.hpp:28]: I think that creation of variable that is returned 2 lines below is not necessary. 
```suggestion
   return  lhs/NWD(lhs, rhs) * rhs;
```

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 5]
```diff
@@ -3,6 +3,13 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    // Would fail for empty vec - but the behaviour is unspecified in task
+    auto currentMax = *vec.begin();
+    for(auto el : vec){
```
Comment (kontrybutor) [inline module2/homework/maxOfVector/maxOfVector.hpp:8]: You can make it ```const &``` as you don't modify the ``el``.

File: module2/homework/nwdNww/nwdNww.hpp (hunk 3 above)
Comment (piezol) [reply module2/homework/nwdNww/nwdNww.hpp:24]: std::abs was committed there through my suggestion on github, i didn't notice the lines below

File: module2/homework/generateSequence/generateSequence.hpp [hunk 6]
```diff
@@ -1,7 +1,15 @@
 #pragma once
 #include <vector>
-
+#include<iostream>
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:3]: Please sort include and add space.
```
#include <iostream>
#include <vector>
```

Please also add enter to separate include file form rest code.

File: module2/homework/generateSequence/generateSequence.hpp [hunk 7]
```diff
@@ -1,7 +1,15 @@
...
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    std::vector<int> vec;
+    if (count <= 0){
+        return vec;
+    }
+    vec.reserve(count);
+    int currentnum = 0;
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:10]: current_num or currentNum

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 8]
```diff
@@ -3,6 +3,13 @@
 #include <limits>
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:3]: Remove if not use

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 9]
```diff
@@ -3,6 +3,13 @@
...
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    // Would fail for empty vec - but the behaviour is unspecified in task
+    auto currentMax = *vec.begin();
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:7]: vec[0] is faster. You don't need to call function begin, to get iterator, and then deference it to get value.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 10]
```diff
@@ -1,11 +1,27 @@
 #pragma once
-
+#include <cmath>
+#include <algorithm>
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:3]: Sort and add enter

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05
+3 XP cooperation

{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635, 1636],
  "points": [2, 4, 2, 6, 2, 5, 2, 5, 3],
  "users": ["piezol", "mefox256"]
}

**Pull request closed without merging**